test/spec/services/angular-apimock.js

```diff
@@ -377,6 +376,45 @@ describe('Service: apiMock', function () {
 				});
 
 			});
+			
+			describe('enable query params', function () {
+
+				beforeEach(function() {
+					apiMockProvider.config({stripQueries: false});
+					// Set global flag to mock
+					$location.search('apiMock', true);
+				});
+
+				afterEach(function() {
+					apiMockProvider.config({stripQueries: true});
+				});
+
+				it('should NOT ignore query objects in request URL (path has /?)', function () {
+					defaultRequest.url = '/api/pokemon/?name=pikachu';
+					defaultExpectPath = '/mock_data/pokemon/name=pikachu.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should NOT ignore query objects in request URL (path has only ?)', function () {
+					defaultRequest.url = '/api/pokemon?name=pikachu';
+					defaultExpectPath = '/mock_data/pokemon/name=pikachu.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should NOT ignore query objects in request OBJECT', function () {
+					defaultRequest.url = '/api/pokemon';
+					defaultRequest.params = {'name': 'pikachu', 'strength': 'electricity'};
+					defaultExpectPath = '/mock_data/pokemon/name=pikachu&strength=electricity.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should sort query objects in alphabetical order', function () {
+					defaultRequest.url = '/api/pokemon';
+					defaultRequest.params = {'strength': 'electricity', 'name': 'pikachu'};
```

Comment:
I've come to the conclusion that we'll need some way to normalize the URI's ourselves, because I can't find anything in Angular to help us (even if the new [$httpParamSerializerJQLike](https://docs.angularjs.org/api/ng/service/$httpParamSerializerJQLike)) in 1.4 seemed promising).